flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -457,4 +486,79 @@ private void registerReader() {
     ListState<SplitT> getReaderState() {
         return readerState;
     }
+
+    private static class LatencyMarkerEmitter<OUT> {
```

Comment:
Yes, I would like to deduplicate this class with `LatencyMarksEmitter` which is in `StreamSource`.

Reply:
@pnowojski , this couldn't deduplicate with the `LatencyMarksEmitter` in `StreamSource`, because this uses the `DataOutput` but `LatencyMarksEmitter` uses the `Output`.